test/unit/redux/navigation.test.js

```diff
@@ -238,4 +247,28 @@ describe('unit test lib/validate.js', () => {
         const resultState = navigationReducer(initialState, action);
         expect(resultState.loginOpen).toBe(false);
     });
+
+    test('handleRegistrationRequested with useScratch3Registration true navigates user to /join, ' +
+        'and does NOT open scratch 2 registration', () => {
+        const initialState = {
+            registrationOpen: false,
+            useScratch3Registration: true
+        };
+        const action = handleRegistrationRequested();
+        const resultState = navigationReducer(initialState, action);
+        expect(resultState.registrationOpen).toBe(false);
+        expect(global.window.location).toEqual('/join');
+    });
+
+    test('handleRegistrationRequested with useScratch3Registration false does NOT navigate user away, ' +
```

Comment:
Yay tests!  What about the case where registrationOpen is set to true and so is useScratch3Registration?  I think it is worth making sure we have a test for that case too.

Reply:
Good idea, done!